tests/conftest.py

```diff
@@ -6,27 +6,56 @@
 import pytest
 
 if TYPE_CHECKING:
-    from typing import Any, Callable, Dict, Literal, Optional, Type, Union
+    from typing import Any, Literal, Optional, Protocol, Union
 
     from requests_mock import Mocker
     from utils import ResourceType
 
     from otelib.client import OTEClient
 
-    OTEResponse = Callable[
-        [
-            Union["HTTPMethod", str],
-            str,
-            int,
-            Optional[Union[Dict[str, Any], str]],
-            Optional[dict],
-            Optional[bytes],
-            Optional[Union[dict, str]],
-            Optional[str],
-            Optional[OTEClient],
-        ],
-        None,
-    ]
+    class TestResourceIds(Protocol):
+        """Return a test id for the given `resource_type`."""
```

Comment:
```
    """
    Defines a protocol for fetching a test ID based on the provided resource type.
    
    This can be used to ensure that the implementing function or method 
    provides a test ID for a given resource type, which can be either 
    an instance of `ResourceType` or a string.
    """
```